edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/booking/BookingStateMachineImpl.java

```diff
@@ -1,12 +1,15 @@
 package org.dcsa.edocumentation.domain.booking;
 
-import org.dcsa.edocumentation.domain.dfa.DFA;
+import org.dcsa.edocumentation.domain.dfa.AbstractStateMachine;
+import org.dcsa.edocumentation.domain.dfa.CannotLeaveTerminalStateException;
 import org.dcsa.edocumentation.domain.dfa.DFADefinition;
+import org.dcsa.edocumentation.domain.dfa.TargetStateIsNotSuccessorException;
 import org.dcsa.edocumentation.domain.persistence.entity.enums.BkgDocumentStatus;
+import org.dcsa.skernel.errors.exceptions.ConcreteRequestErrorMessageException;
 
 import static org.dcsa.edocumentation.domain.persistence.entity.enums.BkgDocumentStatus.*;
 
-public class BookingStateMachineImpl implements BookingStateMachine {
+public class BookingStateMachineImpl extends AbstractStateMachine<BkgDocumentStatus> implements BookingStateMachine {
```

Comment:
what's the benefit of having a separate BookingStateMachine interface and impl? Are we expecting more implementations?

Reply:
Primarily that it was easier to see what was going on (i.e., what methods do the state machine have).  I can collapse them into a single class if you do not feel the (current) implementation is distracting.

Reply:
I think now that other parts have been separated these can be collapsed into a single class.

Reply:
Fixed